tools/verify_translations.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/python3
+# Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+# For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+import collections
+import os
+import os.path as path
+import argparse
+import json
+
+
+def detect_duplicate_keys(list_of_pairs: list):
+    key_count = collections.Counter(k for k, v in list_of_pairs)
+    duplicate_keys = ', '.join(k for k, v in key_count.items() if v > 1)
+
+    if len(duplicate_keys) != 0:
+        raise ValueError(duplicate_keys)
+
+
+def validate_data(list_of_pairs: list):
+    detect_duplicate_keys(list_of_pairs)
+    # More dectection, each of them will raise exception upon invalid
+    # data
+    return dict(list_of_pairs)
+
+
+def check_duplicates(file_path: path):
+    with open(file_path) as json_file:
+        try:
+            _ = json.load(json_file, object_pairs_hook=validate_data)
+        except ValueError as e:
+            dup_list = str(e).split(',')
+            print(f"[{path.basename(file_path)}]: duplicate {len(dup_list)}: {dup_list}")
+
+
+def check_empty_entry(file_name: str, json_data: json):
+    empty_entries = [entry for entry in json_data if len(str(json_data[entry])) == 0]
+    if len(empty_entries) != 0:
+        print(f"[{file_name}]: empty entries {len(empty_entries)}: {empty_entries}")
+
+
+def main(json_path: path):
+    dir_list = os.listdir(json_path)
+    json_keys = []
+
+# iterate to get all possible keys and check for key duplicates
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        check_duplicates(file_path)
+
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            check_empty_entry(file, json_data)
+
+            json_keys.append(set([key for key in json_data]))
+
+    all_keys = set.union(*json_keys)
+
+# iterate to find missing keys
+    for file in dir_list:
+        file_path = path.join(json_path, file)
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+            missing_keys_in_file = all_keys.difference(set([key for key in json_data]))
```

Comment:
List comprehension seems to be redundant in this place. This should work same:
`missing_keys_in_file = all_keys.difference(set(json_data))`

Reply:
fixed